### DISCUSSION TITLE
MinIO fault tolerance with big drives and nodes counts

### DISCUSSION THREAD

**Discussion by AlexZIX** [minio/minio#15026] [Q&A] [answered]
Hi,

I try to understand the real fault tolerance of big minio setups but can't. In sample (https://docs.min.io/docs/distributed-minio-quickstart-guide.html) written:

"For example, an 16-server distributed setup with 200 disks per node would continue serving files, up to 4 servers can be offline in default configuration i.e around 800 disks down MinIO would continue to read and write objects."

But below I've found the notice:

"Each object is written to a single EC set, and therefore is spread over no more than 16 drives."

Is it possible that object becomes unavailable because all its parts located at 16 drives which are subset of 800 drives which goes down? Or minio guarantees that object parts spread over nodes (not drives) evenly? But in this case I also can lost my object in case of just 16 drives failure on 16 nodes, is that right? And also this means that setup with more that 16 nodes are meaningless?

**Comment by klauspost** [comment, marked as answer]
> Is it possible that object becomes unavailable because all its parts located at 16 drives which are subset of 800 drives which goes down? 

Yes. You can lose 4 of 16 disks in a set. In a 16 node setup that is easy, "disk 1" on each server will be part of first set, "disk 2" second set, etc. You can up this by generating more erasure codes - up to 8 for a 16 disk set.

> And also this means that setup with more that 16 nodes are meaningless?

No, it will scale both in terms of size and speed.

**Reply by AlexZIX** [reply]
> Yes. You can lose 4 of 16 disks in a set. In a 16 node setup that is easy, "disk 1" on each server will be part of first set, "disk 2" second set, etc. You can up this by generating more erasure codes - up to 8 for a 16 disk set.

But in this case start command will be

http://minio-{1...16}/disk1 http://minio-{1...16}/disk2 http://minio-{1...16}/disk3 ... http://minio-{1...16}/disk200

Or minio automatically creates internode erasure code set in case of

http://minio-{1...16}/disk{1...200}

?

**Reply by harshavardhana** [reply]
@AlexZIX these discussions are best had on a call with our sales involved feel free to reach out to us at sales@min.io 

We can recommend hardware and size your cluster appropriately.

**Reply by ravindk89** [reply]
@AlexZIX MinIO automatically handles generating erasure code sets  -  and in fact your first syntax woulc create multiple single-disk pools, which would be very inadvisable.

We have general documentation on running MinIO in distributed mode [here](https://docs.min.io/minio/baremetal/installation/deploy-minio-distributed.html). As noted, more specific sizing and hardware recommendations are handled through professional engagement.
